### DISCUSSION TITLE
Configuring a formula to build bottles using the newest version of Xcode available on a macOS runner

### DISCUSSION THREAD

**Discussion by rgoldberg** [Homebrew/discussions#5810] [Writing Formulae/Casks]
### Output of `brew config`

```shell
n/a
```

### Output of `brew doctor`

```shell
n/a
```

### Description of issue

GitHub macOS runners have multiple versions of Xcode installed.

How can I configure a formula to build bottles using the newest version of Xcode available on the macOS runner?

If I cannot specify using the latest, can I specify a specific version?

This is different than setting the minimum Xcode version required to build, as I want the bottles to be built with the latest Xcode, not with a legacy version.

Thanks.

**Comment by SMillerDev** [comment]
If you specify the minimum as the latest version it will take that version.

**Reply by rgoldberg** [reply]
@SMillerDev Thanks for the info.

Setting the minimum Xcode version required to build the formula to be the latest version available on a GitHub macOS runner would require having a different minimum Xcode for each macOS.

It would also prevent users from building from source with an older Xcode on a newer macOS, even if the older Xcode could build mas.

Given those limitations, I’ll look into seeing if our build script can somehow choose the version of Xcode itself, presumably using `xcode-select`. The script could accept an argument specifying using the latest Xcode version, and that argument could be passed in our formula.

I’m just trying to not reinvent the wheel, so if Homebrew already offered support for something like this, I would have preferred to use it instead of rolling my own version.

Thanks again.
